atc/worker/gardenruntime/worker.go

```diff
@@ -715,11 +715,33 @@ func (worker *Worker) findOrStreamVolume(
 				return Volume{}, err
 			}
 
-			delegate.StreamingVolume(logger, inputPath, artifact.Source(), streamedVolume.DBVolume().WorkerName())
+			// if volume size not cached, get source volume size through LookupVolumeWithSize
+			var bcSourceVolume baggageclaim.Volume
+			if volume.DBVolume().Size() <= 0 {
+				bcSourceVolume, _, err = volume.worker.bcClient.LookupVolumeWithSize(ctx, volume.DBVolume().Handle())
+				if err != nil {
+					logger.Error("failed-to-find-source-volume-for-streaming", err)
+					return Volume{}, err
+				}
+				if inputPath == "for image" {
```

Comment:
Based on the metrics, streamed image volume accounts for more than 80% of the total streamed volume(image & non-image). 
Also, calculate&cache image volume size should make more sense than non-image volume, because non-image volume size may change swiftly but image volume size could be considered as unchanged.